src/openlcb/DccAccyProducer.hxx

```diff
@@ -0,0 +1,250 @@
+
+    /// Destructor.  Remove "this" instance from @ref instances_ vector
+    ~DccAccyProducer()
+    {
+        AtomicHolder h(this);
+        for (unsigned i = 0; i < instances_.size(); ++i)
+        {
+            if (instances_[i] == this)
+            {
+                instances_.erase(instances_.begin() + i);
+                break;
+            }
+        }
+    }
+
+private:
+    using Command = DccAccyProducerInput::Command;
+
+    /// Entry point to sub-flow that dispatches the next state based on the
+    /// incoming command.
+    ///
+    /// @return next state appropriate to the command, else
+    ///         Defs::ERROR_INVALID_ARGS on error
+    Action entry() override
+    {
+        HASSERT(input()->address_ > 0 && input()->address_ <= MAX_ADDRESS);
+        switch (input()->cmd_)
+        {
+            case Command::CMD_QUERY:
+                return call_immediately(STATE(query));
+            case Command::CMD_SET:
+                return call_immediately(STATE(set));
+            default:
+                return return_with_error(Defs::ERROR_INVALID_ARGS);
+        }
+    }
+
+    /// Query the last known state of the accessory address.
+    ///
+    /// @return waits for the query message to go out and advances to the
+    ///         parent flow's next state once the memory holding the message is
+    ///         freed.
+    Action query()
+    {
+        eventProducer_->send_query_consumer(input()->address_ - 1, &writer_,
+                                            bn_.reset(this));
+        return wait_and_return_ok();
+    }
+
+    /// Set the state of the accessory address.
+    ///
+    /// @return waits for the query message to go out and advances to the
+    ///         parent flow's next state once the memory holding the message is
+    ///         freed.
+    Action set()
+    {
+        eventProducer_->set(input()->address_ - 1, input()->value_, &writer_,
+                            bn_.reset(this));
+        return wait_and_return_ok();
+    }
+
+    /// Helper method for accessing the subflow input data
+    ///
+    /// @return pointer to the subflow data
+    DccAccyProducerInput *input()
+    {
+        return message()->data();
+    }
+
+    /// Callback called when there is a notification of event (DCC accessory
+    /// address) state.  Pass to the next level up after accounting for DCC
+    /// address range starting at 1.
+    ///
+    /// @param bit bit index within event pair range
+    /// @param value value of the event pair
+    static void state_callback(unsigned bit, bool value)
+    {
+        for (DccAccyProducer* instance : instances_)
+        {
+            if (instance->dccStateCallback_)
+            {
+                instance->dccStateCallback_(bit + 1, value);
+            }
+        }
+    }
+
+    /// Singleton instance of a BitRangeNonAuthoritativeEventP for the DCC
+    /// accessory address Well-Known Event ID range.
+    static BitRangeNonAuthoritativeEventP *eventProducer_;
```

Comment:
Can this member be statically constructed in place rather than using operator new at runtime?  This would save a malloc().  Is there a construction order problem with this?